File: tests/fixtures/source-windows/haskell/module-functions.hs
module Windows.Report (Entry(..), render) where

data Entry = Entry
  { title :: String
  , owner :: Maybe String
  }

render :: [Entry] -> [String]
render entries = map renderEntry entries
  where
    renderEntry entry =
      let resolvedOwner = maybe "unassigned" id (owner entry)
      in title entry ++ " — " ++ resolvedOwner

visible :: [Entry] -> [Entry]
visible = filter (not . null . title)

File: tests/fixtures/source-windows/elixir/module-functions.ex
defmodule Windows.ReportBook do
  @moduledoc "Keep report entries in insertion order."

  defstruct entries: []

  def new, do: %__MODULE__{}

  def add(book, title, owner) do
    entry = %{title: title, owner: owner}
    %{book | entries: book.entries ++ [entry]}
  end

  def render(book) do
    Enum.map(book.entries, fn entry ->
      owner = entry.owner || "unassigned"
      "#{entry.title} — #{owner}"
    end)
  end

  def size(book), do: length(book.entries)
end
